Update typescript.md (#1495)
cap-js/docs: node.js/typescript.md

> ## TypeScript APIs in `@sap/cds`

Reviewer: ```suggestion
## TypeScript APIs in `@sap/cds` <Since version="8.0.0" of="@sap/cds" />
```

Reviewer 2: Awesome, thanks for making these amendments! 🙂

> ~The package `@sap/cds` is shipped with TypeScript declarations~ [See release notes](https://cap.cloud.sap/docs/releases/jun24#cds-types-dependency). These declarations are used automatically when you write TypeScript files, but also enable IntelliSense and type checking for standard JavaScript development in Visual Studio Code.

Reviewer: I'll make that code a codeblock later. That doesn't work in suggestions.

```suggestion
The package `@cap-js/cds-types` contains all TypeScript declarations for `@sap/cds` APIs. These declarations are used automatically when you write TypeScript files, but also enable IntelliSense and type checking for standard JavaScript development in Visual Studio Code. Just add the `@cap-js/cds-types` package to your project as follows:

`npm add @cap-ja/cds-types`

```

Reviewer 2: Reads much better! I'd just like to add for consideration that cds7 actually still comes with type-batteries included. "Only" users of cds8 have to manually install cds-types. But I believe CAPire always targets the latest version anyway?

Reviewer: Yes, we target always the latest version. What would happen if a project on cds7 installs the cds-types package standalone?

We could add a badge to that section for clarity, like `<Since version="8.0.0" of="@sap/cds" />`. But I'd only like to do it if a cds7 user breaks his project using the direct dependency.

Reviewer 2: The types users would see from explicitly installing cds-types at its latest version alongside a cds7 project would probably show the API for cds8. This isn't a problem for parts of cds that have not changed between v7 and v8. While it likely wouldn't _break_ their project, it could lead to confusion when users try to use cds8 API in their cds7 project, which then crashes on runtime, where that API is not actually present.

Reviewer: Alright, then let's do it :)

> Use them like this:

Reviewer: ```suggestion
Use the Typescript declarations like this:
```
